From 848c9679965cd334477db293fc21bd08ce60f685 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:37 -0400
Subject: [powerpc] module: Fix stubs for BE

Message-id: <1410545655-205645-529-git-send-email-dzickus@redhat.com>
Patchwork-id: 94405
O-Subject: [RHEL7 PATCH 528/626] powerpc/module: Fix stubs for BE
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 872aa779bcdd26bbe2d538c36e2c2072b3f03c8c
Author: Rusty Russell <rusty@rustcorp.com.au>
Date:   Mon May 12 17:15:02 2014 +0930

    powerpc/module: Fix stubs for BE

    A simple patch which was supposed to swap r12 and r11 also
    inexplicably changed the offset by two bytes.  This instruction
    (to load r2) isn't used in LE, so it wasn't noticed.

    Fixes: b1ce369e82 ("powerpc: modules: use r12 for stub jump address.)
    Reported-by: Alistair Popple <alistair@popple.id.au>
    Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
    Tested-by: Alistair Popple <alistair@popple.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/module_64.c b/arch/powerpc/kernel/module_64.c
index ef349d0..077d2ce 100644
--- a/arch/powerpc/kernel/module_64.c
+++ b/arch/powerpc/kernel/module_64.c
@@ -134,7 +134,7 @@ static u32 ppc64_stub_insns[] = {
  0xe98b0020,   /* ld      r12,32(r11) */
 #if !defined(_CALL_ELF) || _CALL_ELF != 2
  /* Set up new r2 from function descriptor */
- 0xe84b0026,   /* ld      r2,40(r11) */
+ 0xe84b0028,   /* ld      r2,40(r11) */
 #endif
  0x7d8903a6,   /* mtctr   r12 */
  0x4e800420   /* bctr */
-- 
1.7.1